pkgs/development/python-modules/ninja/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchurl
+, cmake
+, setuptools-scm
+, scikit-build
+, pytestCheckHook
+, pytest
+, pytest-cov
+, pytest-runner
+, pytest-virtualenv
+, codecov
+, coverage
+}:
+buildPythonPackage rec {
+  pname = "ninja";
+  version = "1.11.1";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "scikit-build";
+    repo = "ninja-python-distributions";
+    rev = version;
+    hash = "sha256-scCYsSEyN+u3qZhNhWYqHpJCl+JVJJbKz+T34gOXGJM=";
+  };
+  patches = [
+    # make sure cmake doesn't try to download the ninja sources
+    ./no-download.patch
+  ];
+
+  # see NinjaUrls.cmake
+  ninja_src = fetchurl {
```

Comment:
Should we assert that the version matches?

Reply:
I've added a check based on the hash